js/ui/data_grid.d.ts

```diff
@@ -1444,2238 +1303,3296 @@ export interface GridBaseOptions<T = GridBase> extends WidgetOptions<T> {
+     * @default null
+     * @action
+     * @prevFileNamespace DevExpress.ui
+     * @public
+     */
+    onFocusedRowChanging?: ((e: FocusedRowChangingEvent) => any);
+    /**
+     * @docid
+     * @type_function_param1 e:object
+     * @type_function_param1_field4 event:event
+     * @type_function_param1_field5 data:object
+     * @type_function_param1_field6 key:any
+     * @type_function_param1_field7 values:Array<any>
+     * @type_function_param1_field8 columns:Array<Object>
+     * @type_function_param1_field9 rowIndex:number
+     * @type_function_param1_field10 rowType:string
+     * @type_function_param1_field11 isSelected:boolean
+     * @type_function_param1_field12 isExpanded:boolean
+     * @type_function_param1_field13 isNewRow:boolean
+     * @type_function_param1_field14 groupIndex:number
+     * @type_function_param1_field15 rowElement:dxElement
+     * @type_function_param1_field16 handled:boolean
+     * @default null
+     * @action
+     * @prevFileNamespace DevExpress.ui
+     * @public
+     */
+    onRowClick?: ((e: RowClickEvent) => any) | string;
+    /**
+     * @docid
+     * @type_function_param1 e:object
+     * @type_function_param1_field4 event:event
+     * @type_function_param1_field5 data:object
+     * @type_function_param1_field6 key:any
+     * @type_function_param1_field7 values:Array<any>
+     * @type_function_param1_field8 columns:Array<dxDataGridColumn>
+     * @type_function_param1_field9 rowIndex:number
+     * @type_function_param1_field10 rowType:string
+     * @type_function_param1_field11 isSelected:boolean
+     * @type_function_param1_field12 isExpanded:boolean
+     * @type_function_param1_field13 isNewRow:boolean
+     * @type_function_param1_field14 groupIndex:number
+     * @type_function_param1_field15 rowElement:dxElement
+     * @default null
+     * @action
+     * @prevFileNamespace DevExpress.ui
+     * @public
+     */
+    onRowDblClick?: ((e: RowDblClickEvent) => any);
+    /**
+     * @docid
+     * @type_function_param1 e:object
+     * @type_function_param1_field4 data:object
+     * @type_function_param1_field5 key:any
+     * @type_function_param1_field6 values:Array<any>
+     * @type_function_param1_field7 columns:Array<dxDataGridColumn>
+     * @type_function_param1_field8 rowIndex:number
+     * @type_function_param1_field9 rowType:string
+     * @type_function_param1_field10 groupIndex:number
+     * @type_function_param1_field11 isSelected:boolean
+     * @type_function_param1_field12 isExpanded:boolean
+     * @type_function_param1_field13 isNewRow:boolean
+     * @type_function_param1_field14 rowElement:dxElement
+     * @default null
+     * @action
+     * @prevFileNamespace DevExpress.ui
+     * @public
+     */
+    onRowPrepared?: ((e: RowPreparedEvent) => any);
+    /**
+     * @docid
+     * @type boolean|object|Enums.Mode
+     * @default "auto"
+     * @prevFileNamespace DevExpress.ui
+     * @public
+     */
+    remoteOperations?: boolean | {
+    /**
```

Comment:
Why is the remoteOperations object is not extracted into a separate type?

Reply:
Because when we discuss improvements with Ilya, we decided that such types we skip now and will extract later.